### PR TITLE
Release v5.0.0

### LINKED ISSUE #276: Feature request: Allow language switching
Please provide an easy way to switch the interface language. The language could be selected from drop down menu in Mapeo-mobile interface.

In SE. Asia the phone default language set to English even though native language of user is Khmer or Myanmar. So the default locale doesn't accurately represent the users language choice.

### LINKED ISSUE #293: Allow importing custom default settings from settings UI
Deploying customized icons/profiles aka. [default setting](https://github.com/digidem/mapeo-default-settings) requires physical access to target mobile phone using ADB tool chain and a skilled technical person on site to do this.  

Perhaps a simple solution would be import profile setting in Mapeo-mobile setting UI which enables the user to import customized default settings from the mobile device itself. The customized default settings could be sent to mobile device via a email or social media messaging.

### LINKED ISSUE #336: Vietnamese Translation Complete
Please kindly merge the Vietnamese localization strings. Thank you @nga-nguyen-vn for all the work!

### PR DESCRIPTION
**Pull request by gmaclennan**
This PR is to prepare for release v5.0.0 (previously v2.1.0, but bumping the version number to bring in-line with Mapeo Desktop). This is testing a new release process, where "Release Candidates" for testing will be shared in this PR. When it has passed QA testing then this will be merged with the `deploy` branch and released to Google Play, Github Releases, etc.

See the [draft changelog](https://github.com/digidem/mapeo-mobile/blob/release/v2.1.0/CHANGELOG.md) for changes since our last public release (v2.0.0-beta.5)

Note that previously we released "betas" publicly. From now on we no longer use "beta" and "alpha" terms.

**APKs for testing**
- [QA v5.0.0-RC.844](https://mapeo-apks.s3.amazonaws.com/beta-testing/v5.0.0/mapeo-v5.0.0-RC.844-QA-debug.apk): You can install this alongside your existing Mapeo App and test out new features without affecting your existing data. Look for the red logo.
- ⚠️[v5.0.0-RC.844 APK](https://mapeo-apks.s3.amazonaws.com/beta-testing/v5.0.0/mapeo-v5.0.0-RC.844-universal.apk): This will replace your existing Mapeo installation and you will not be able to downgrade - if it does not work you will need to wait for a fix.

### CODE REVIEW

**Comment by okdistribute** [comment]
Great work! What's RC.760 stand for?

**Comment by gmaclennan** [comment]
“RC” is for “Release Candidate”. As for the number “760”, I wanted to automate builds on each push to the PR branch, and I wanted an incrementing number so we know what the latest release candidate is, so 760 is the build number (a number that increments every time CI runs). We also use the build number as the “version code” of the app - every android app needs an integer version code and it must always increment - you cannot downgrade to a lower version code ever. When I finish the GitHub actions, every push to a release PR will leave a comment with a link to a new release candidate build. For now I need to do that manually if QA finds any bugs with this RC.

**Comment by gmaclennan** [comment]
New Release Candidate build 762 for testing. Fixes:

* Move attribution (i) to bottom-right (avoid overlap with scale bar)
* Fix crash during sync progress (updates progress circle library)

**APKs for testing**
- [QA v2.1.0-RC.762](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.762-QA-debug.apk): You can install this alongside your existing Mapeo App and test out new features without affecting your existing data. Look for the red logo.
- ⚠️[v2.1.0-RC.762 APK](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.762-universal.apk): This will replace your existing Mapeo installation and you will not be able to downgrade - if it does not work you will need to wait for a fix.

**Comment by aliya-ryan** [comment]
You have asked me to review this PR but I don't know what review means. Is that to do with testing something - obvs. you don't want me to look at the code but happy to do whatever it is I can.

**Comment by gmaclennan** [comment]
Hi @aliya-ryan sorry for the confusion, I'm still figuring out the details of how to improve this release process and document it all.

The idea of assigning you and @jencastrodoesstuff as reviewers of this PR is to have a mechanism for recording sign-off on the new release. What this means for you and Jen is to test the Mapeo Release Candidate, which is linked in the [issue above](https://github.com/digidem/mapeo-mobile/pull/358#issuecomment-610298413). Once you have tested it you can add your review by clicking "Add Review" at the top of this page. If it the app works well and you think it is ready for release, you can select "Approve" and then "Submit review". If you think something is broken and needs fixed, then you can select "Request changes" and you can add a description of what is not working, and submit your review.

Merging this PR will trigger this version of Mapeo Mobile (v2.1.0) to be released†, which means that it will become available to download on our website and on the Google Play Store. But this PR cannot be merged until both you and Jen have approved it.

This way we can have a clear sign-off process on releases, and the process is clearly communicated and anybody can click "Squash and Merge" to actually create the release once all the reviews are approved (e.g. reduced dependency on me).

So, in summary what I am requesting from you is to download the Mapeo build linked above, test it works ok, and approve this PR if it does. Please let me know how this process works for you.

†This isn't actually true yet, I haven't automated all steps in the process yet.

**Comment by jencastrodoesstuff** [comment]
@gmaclennan I appreciate the work going into reducing dependency on you. 
Long term this will surely help. I very much understand that building the structures (this release page) and asking that we actively participate in the process to prevent releases with untested issues going into the world makes sense. But this is all extremely new. 
We have been referencing https://www.notion.so/digidem/Release-Process-REQUIRES-REVISION-353c3c0d97ec47078c3e6218bccb67e0 for the process and already noted that things need updating... It feels like we are testing a testing process which is ok. but it is very disorienting to go through this without having an idea of what high level goals and checkboxes we are addressing.
Testing scripts are being updated slowly. I can focus on that for the next while as I execute testing on Mm & Md. Knowing if they are actually useful in identifying the things the tech team needs to know is the second step for me. The where and how to put them may be here. but is the issue resolution process really best here? If approval from me may in the future initiate a build for release to be made then our testing scripts need to be thorough and reviewed by the tech team. 
We are trying to eliminate human errors in technology as humans with technology... Tricky.

**Comment by aliya-ryan** [comment]
Test of release on Samsung Galaxy Prime+ and Ulefone Armor 3T

The GPS button in Mapeo did not display correctly on either phone initially. 
On the Ulefone the first time I opened it, the button remained in searching mode, even when it had found GPS coordinates. Once I pressed + to add a point, from then on it displayed the accuracy properly. On the Samsung it didn't change from 'searching' even when it had a GPS coordinates as evident on the GPS details screen and on adding a point, even after taking lots of points. There was no difficulty adding a point with it on 'searching' mode, and the point then appeared in the right place, and the blue dot was in the right place, but the 'searching' was confusing and misleading.
- [ ]  Much later when I was testing something else on the Samsung, I turned the GPS location button on and off on the phone, and after that it did display the accuracy correctly within Mapeo

**Comment by aliya-ryan** [comment]
I deleted the maptiles off phone to see what would happen. With no maptiles installed, and original mapbox map folder deleted. It has a white screen with "error loading map", and none of the points appear. Can still seem to take points, but the map face remains totally white and the scale does not change if pinch. GPS button works correctly without map.
As this is a very likely thing that might happen for partners, it would be good if a screen appeared telling them what was going on and that they could still collect data - would also be good if the points could still appear even if no map behind them (no way of getting a very very undetailed map in as default so that people could see they were at least in the right country??).

**Comment by aliya-ryan** [comment]
If the GPS of the phone is switched off and then you try and take a point, in the place where the coordinates would normally appear it says "buscando" where it would be better it would say "Sin GPS" like the GPS button does, or GPS device switched off, or something.

**Comment by aliya-ryan** [comment]
The testing script says there should be a shutter sound when you take a photo. I can't remember if I've heard one before or not. But neither phone had a shutter sound when photo was taken.

**Comment by aliya-ryan** [comment]
If a photo was taken landscape on the Ulefone then it did not always turn it to landscape as expected. Usually it did, but on a couple of occasions it didn't. It may be that I righted the phone very quickly after taking photo, but photo looks fine, just wrong way round.
Might we be able to add a "turn photo" button anywhere at some point?

**Comment by aliya-ryan** [comment]
Sharing:
- I tried to share with gmail on a phone where it didn't already have my account set up. Once I realised and try to go back this didn't work, and I had to quit app and reopen to get back to Mapeo.
- Sharing on a phone where I did have gmail set up worked as I would expect.
- Sharing via bluetooth to my computer from both phones was unsuccessful.
- Sharing via whatsapp works, but if there is more than one photo then it sends one message with all info for each photo.
- I shared with google maps - it only seems to recognise the photo info and you have to them tell it where to place it (on google maps). I don't know if it worked (photo is not there but perhaps it gets reviewed).

**Comment by aliya-ryan** [comment]
Details screen - dot does not appear on map in expected place - map is there and the pointer points to right place but I would expect an orange dot above the pointer.

**Comment by aliya-ryan** [comment]
Collecting details
- If there are lots of questions in the list then you have to go through all of them before you can save. Would be good if there was a shortcut to save so that you coudl get out of it quickly.

**Comment by aliya-ryan** [comment]
Photo delete
I thought - but perhaps i remember wrongly - that you used to be able to delete photographs after taking them, but this isn't possible now. Maybe it never was.

**Comment by aliya-ryan** [comment]
Editing observations
If you edit an observation, and then press the go back button rather than the save button, the message says "discard observation" / descartar... when really it should say discard edits as you do not delete the whole observation by not saving changes.

**Comment by aliya-ryan** [comment]
Details 
The little blue lightening bolt that appears in top right of screen in details of points taken on another phone. I understand what it means but I don't think it's obvious - maybe it could be a button and if pressed it explained why you can't edit that point.

**Comment by aliya-ryan** [comment]
Aliya tested Mm release. Having problems with the settings loader, and also with the sync to computer. But everything else tested and any issues I found reported above.

**Comment by aliya-ryan** [comment]
Settings loader does not work on either phone - it takes me to a screen allowing me to choose the file, but when i choose it simply returns to the old screen showing old configurations. If I load settings the old way by copying in folder it works.

**Comment by aliya-ryan** [comment]
Project Key does not load correctly.
On both the configurations detail screen and on the sync screen the Project Key is listed as MAPEO. This does not change if you change the configurations, and I know it is not correct for the configurations currently within Mapeo.
Perhaps this is why I can't get it to be recognised on Mapeo Desktop?

**Comment by aliya-ryan** [comment]
Mm not picked up by MD on sync

Tried multiple times, quitting Mapeo D and Mm, trying on different networks (hotspot and TP link), trying with different versions of the settings installed, but I can not get the phone and computer to pick each other up. Using Mapeo 5.1.0 on desktop - I had also tried with Mapeo 5.0.0 with similar failure.

My only guess currently is that this is to do with the project key not being processed correctly on the phones and hence not matching the one on the computer (see previous comment).

**Comment by okdistribute** [comment]
Documenting a bug reported by Jen today, with Md 5.1.0 -- could be a bug in Mapeo Core, although it doesn't seem entirely critical, it is pretty strange. https://github.com/digidem/mapeo-desktop/issues/328

**Comment by okdistribute** [comment]
Found another sync bug that may not be new with this version, but I figured I'd report:

1. Take observations with many photos on phone, enough so sync will take at least a few seconds with progress
2. Open desktop sync
3. Open phone sync
4. Start sync 
5. Close desktop app
6. See an error (expected):
Screenshot_20200417-144915_Mapeo QA

7. Open Mapeo Desktop again, try to syncronize

Mapeo gets stuck in this state on the phone, but the Desktop app seems to continue just fine and finishes. You can see the photos and everything looks good on the Desktop
Screenshot_20200417-145041_Mapeo QA

9. Close the sync screen on mobile and try to find the Desktop app again --> the mobile app is stuck in a 'searching...' state. I'm not able to get them to find eachother again until I "force quit" the application on the phone and restart it. Then they are able to find each other.

**Comment by gmaclennan** [comment]
@aliya-ryan thanks for the detailed testing. Several of these bugs also exist in previous releases and I suggest we go ahead with this release without resolving them, and work to fix them in the v2.2 release or later. I have created issues for each of them:

- Add option to delete photos from an observation #384 (this has never existed)
- Allow saving an observation without navigating all questions #383
- Show explanation of synced observations #382 
- Cancelling an edit should show "discard edits" #381 
- Sharing an observation does not work consistently #380 
- Include a default offline map #379
- GPS pill in header does not update correctly #378

I think the issues that do need to be resolved for this release are:

- [x] Taking a photo landscape sometimes results in photo being portrait (I think I fixed this and we can check in the next release candidate)
- [ ] Settings loader does not work correctly
- [ ] Project key is not updating after changing settings

The sync errors reported by @jencastrodoesstuff / @okdistribute I have seen before, and I think it's a bunch more work to fix. I suggest moving forwards with this release without fixing this, and work on a fix in the next release.

> Details screen - dot does not appear on map in expected place - map is there and the pointer points to right place but I would expect an orange dot above the pointer.

@aliya-ryan Is what you see different than this?

screen

> The testing script says there should be a shutter sound when you take a photo. I can't remember if I've heard one before or not. But neither phone had a shutter sound when photo was taken.

The shutter sound is hard-coded into some phones. This is because of the horrible issue of people taking "upskirt" photos in public, and as a result some governments require that phones always make a shutter sound when taking a photo. This means that some phones purchased in certain countries will have a shutter sound, and others will not. We could add our own shutter sound in Mapeo but currently we do not.

**Comment by aliya-ryan** [comment]
Thanks for responses @gmaclennan Gregor,

In relation to the question about details screen. On neither phone did I see the orange dot or the little marker. Just the bubble with UTM coordinates underneath. As in images below. 

I also get this error message repeatedly coming up on both phones. 

IMG-8556
IMG-8555
IMG-8554

**Comment by jencastrodoesstuff** [comment]
@aliya-ryan those is a fascinating series of bugs. I'm not experiencing all of them. 

1. Both Samsungs are showing the orange point location icon on the observation screen. 
2. Settings loader - I had some difficulty accessing .mapeosettings files because "internal memory" (from the left side menu) showed it was empty so I had to go to "Downloads" and then run a search for the folder I pushed into the documents folder I named MAPEO-config-files. Once I selected one of those I could switch configurations, config names and project keys updated on the config setting screen, and I could collect new points and review observations with the new config category names. @gmaclennan is this something to do with app permissions to access files? 

Your other observations are consistent with mine. 
For some of the UX enhancements, I have been adding them as pink highlighted bullets called "Possible desired behavior" within the [testing script ](https://www.notion.so/digidem/Script-for-MAPEO-Mobile-QA-UPDATE-IN-PROGRESS-56fa80375bb94740913b002bf07c1609) to be collected and discussed since they are non-critical.  Would it make sense to gather them all there until the next TP call and, flag and report the urgent UX issues here? Would that help the triage process @gmaclennan ?

**Comment by gmaclennan** [comment]
@jencastrodoesstuff and @aliya-ryan could you both confirm which of the two linked variants you are testing with? I.e the QA variant (red icon) or the regular variant (dark blue icon)? Trying to figure out if the differences are due to that.

@jencastrodoesstuff regarding other UX issues, I'd like to keep this thread to only urgent issues that might block us releasing this to partners. It contains many fixes which are an improvement on what is in the past release, and I'd like to avoid adding new things (e.g. improving some UX issues) which might introduce additional bugs and delay the release.

Regarding permissions for files, that has been very tricky to get right with Android. Unfortunately Android has made changes to the way apps can access the filesystem of the phone in every release of Android, and to make matters more complicated different manufacturers modify Android in different ways. Could @aliya-ryan and @jencastrodoesstuff report the Android versions that your test phones are running?

**Comment by aliya-ryan** [comment]
@gmaclennan I am using the QA version for the testing that came up with the above series of bugs

On Ulefone it is using Android version 8.1.0

On Samsung is using version 6.0.1

**Comment by aliya-ryan** [comment]
@gmaclennan in relation to the map icon dot not appearing - on both the Samsung and the ulefone with the new apk you sent the dots are now appearing as one would expect.

**Comment by jencastrodoesstuff** [comment]
didn't see that we are now on 
QA v2.1.0-RC.775

I was testing RC.762
The RC version # seems important. My red QA version of Mm has been on my phones and updating in similar ways to my standard (blue) release versions so the #'s matter to me. 

Using Android versions: 10 & 9

**Comment by gmaclennan** [comment]
@jencastrodoesstuff not to worry about it, I just updated it with a couple of small fixes for the bug Aliya was seeing, but didn't see the need to notify you until I have added some more fixes which I'm working on right now.

Can you confirm which you were using (red or blue) when doing the testing?

**Comment by jencastrodoesstuff** [comment]
Since this process has been introduced I only test with red

File: src/backend/server.js
Comment (okdistribute) [inline src/backend/server.js:381]: can we put this in mapeo/core v8 branch for v8.1.4?

Comment (gmaclennan) [reply src/backend/server.js:381]: Yes, would support that!

**Comment by jencastrodoesstuff** [comment]
— *MAPEO Issue* —
May 5, 2020, 06:15 PM
UTM 18T 607894 5043333

One icon is not updating even though the data folder contents omly has the new version of the icon with the same file name.

_Issue Priority:_
Minor inconvenience
_Issue Reproducibility:_
I'm not sure how
_Issue Context:_
Updating different builds of this config file with changes to the dd-mapeo-bug icon. Using the config importer.
Preset folder was deleted a number of times and the configuration imported a few times. All updated icons changed except for this one
_Expected Behaviour:_
That the updated build with the updated icon would not display the old icon. Just the new one
_Equipment & OS:_
Galaxy  s10e android 10



_Observer:_
Jen
_Affiliation:_
Dd

5b09679093db42c08177144924c28814
e3f0ae009c445f9c5d3f27ff15f7f29b

— Sent from Mapeo —

**Comment by jencastrodoesstuff** [comment]
— *MAPEO Issue* —
May 5, 2020, 06:44 PM
UTM 18T 607887 5043326

Warning screen on Mm

_Issue Type:_
Error Message,Camera issue
_Issue Priority:_
Minor inconvenience
_Issue Reproducibility:_
I'm not sure how
_Issue Context:_
Creating a new observation with new configurations imported last night
After selecting category a warning message appears

_Expected Behaviour:_
No warning, no issues with previously functioning features
_Equipment & OS:_
Galaxy s10e android 10
_Observer:_
Jen
_Affiliation:_
Dd

— Sent from Mapeo — 

8932af89e7166709150a57c058bb47a7
204561ddd5957c390a4ea82426cf1c13

— Sent from Mapeo —

**Comment by jencastrodoesstuff** [comment]
— *MAPEO Issue* —
May 5, 2020, 08:16 PM
UTM 18T 607885 5043332

Config loading not working on older android 7
The mapeosetting package was not a visible option on in the downloads folder where they were transfered successfully from mac android file transfer

The mapeosetting package was not a visible "my files" app either
They do appear in ES File explorer App

Using google drive link the mapeosettings packages were visible via the Mm file picker. After selection the was a greyed text delay and the it jumped back to normal colouring but the was no settings change
Restarted to test again 3 x and still not working.

_Issue Type:_
Configuration loading issue
_Issue Priority:_
Major issue
_Issue Reproducibility:_
Yes
_Issue Context:_
Newly installed Mm v2.1.0 rc 775
Mapeosettings package transfered to download folder of device memory. (No sdcard)
Initial start up normal.

_Expected Behaviour:_
Load config packge
Confirm new configurations
New categories appear after hitting capture
_Equipment & OS:_
Samsung j5 android7
_Observer:_
Jen
_Affiliation:_
Dd

— Sent from Mapeo — 
[(video link)](https://drive.google.com/file/d/15yyXQOLza7S6AvOnegpUvu-puhhiS74G/view?usp=sharing)  - was made for Kelly until I realized that my google drive workaround didn't work either


4d115cda5d5c7511c80484438b754af4


d1847c6d779429a9964faf3904091046


4c68746d0530535edd599a089c4d0956
56e32ba1ab2c08d362d57ae85228f32c
16de9a79ed1fc669ececb67d1d8a3069

**Comment by gmaclennan** [comment]
@jencastrodoesstuff thanks for the detailed reports.

1. Not seeing the file in downloads.

This is weirdness in older versions of Android and _some_ devices. The issue is that the "Downloads" folder that is visible from apps is not the same as the "Downloads" folder that you see from Android File Transfer. I do not know a workaround, other than downloading via an email / google chrome (and it will go to the "correct" downloads folder) then you should be able to see it from the app. I don't know whether putting it in a different folder will work.

2. Presets not actually updating

I managed to debug an issue with samsung phones today, and the fix in the [latest build RC.781](https://mapeo-apks.s3.amazonaws.com/beta-testing/v2.1.0/mapeo-v2.1.0-RC.781-QA-debug.apk) might fix it if your error was caused by the same bug. If not I'd like to troubleshoot directly on your phone over zoom soon.

3. "Yellow box" warnings

This is helpful seeing these. However these do not show on the version we release and they are just warnings "this _might_ be a problem at some point" but they I think they are all ok issues at the moment.

4. New icon not appearing

I would be interested to know if the updated icon did eventually appear. It could be a caching issue. If it has appeared after 24 hours, then I will see if I can add a fix to skip the cache somehow.

**Comment by jencastrodoesstuff** [comment]
quick
 1. Damn that's anoying. sd's that aren't sd's, downloads that and not the
same as downloads.  Android must be a nightmare to program
2. Thanks!! want to test this today
3. I'll try to get better at tracking when these appear
4. still not showing. on my s8 the right icon does appear. Very strange.
Ghosts


*Jen Castro*
*(Remote from Montreal)*
*Program Coordinator - **Digital Democracy*
Tel Perú:  +51(1)7007968 , Canada: +16479455252 , WhatsApp: +17789984058,
Skype:  live:jen_5964

**Pido disculpas por cualquier error gramática u ortográfica. Estoy en el
proceso de mejorar mi nivel de castellano y aprecio su paciencia.

File: src/backend/server.js
Comment (okdistribute) [inline src/backend/server.js]: would this be storages.length + 2?

one for indexDb and the other for osm.core._logs.close?

Comment (gmaclennan) [reply src/backend/server.js]: Thanks for the catch @okdistribute! I forgot to modify this when I added _logs.close. I've started to lean towards async/await for a lot of things, bugs like this and unhandled errors are too easy to miss with callback patterns.

Comment (gmaclennan) [inline src/backend/server.js:361]:
```suggestion
      let pending = storages.length + 2;
```

**Comment by okdistribute** [comment]
Bug Report: https://github.com/digidem/mapeo-mobile/issues/415

**Comment by gmaclennan** [comment]
@aliya-ryan @jencastrodoesstuff @okdistribute Do you think bug #415 (can't sync after peer disconnects and re-connects) should block this release? I.e. can we release with this bug, or should we try to fix it first?

**Comment by aliya-ryan** [comment]
**Test results** 
_Samsung Galaxy Grand Prime+
Android v 6.0.1_
Full notes in script: https://www.notion.so/digidem/9-July-2020-mapeo-v5-0-0-RC-810-QA-debug-apk-0474cf2d693d4a4cb4da2c7fa7c89c1d

I just did minimal testing of most essential parts - according to shortened script created by @jencastrodoesstuff  for Tom R. to test.

I found no errors I would consider to be blocking (but did not do full test)

I tested sync with my mac, using v5.3.0-beta.2
I could not test sync with another phone as gave my second phone to Tom R. to use for testing

**Comments:**
UI
- When start up the QA, it appears with the blue background and Mapeo logo, would make more sense if the QA could start up with red background instead, to help differentiate.
- The blue dot on the map screen showing current location appears underneath the orange dots of points - I think it would make more sense the other way round
- When I started up offline, it said 'error loading map' - perhaps more useful to say something more instructional like "error loading map, - try going online or load an offline map"
- In the language chooser it might make sense if the current chosen language appears at top of list, perhaps with font/colour differentiation or something.
- When go to sync screen without any wifi network - the Wifi says: Wifi: <unknown ssid> - this isn't very useful, could it not say "device not connected to a wifi network" or "connect to a wifi network to syncronise" or something like that.
- The script says there should be an optoin to "open settings" to go to the wifi menu - there is no "open settings" button - maybe the UI has changed @jencastrodoesstuff do we need to update the script?

Bugs / Issues
- When I opened Mm, it was not connected to internet. The GPS details button said 'searching' but if clicked on it then the points had already been captured. Stayed like this for ages, until I connected to internet - could this be something to do with there not being a map? I could not reproduce this, despite opening and closing Mapeo a few times. - not blocking
- the online map didn't load once I connected to internet until I had quit and reopened Mapeo - not blocking
- When syncing with computer, for a brief moment on the phone it came up with an error message saying something like "Error in Mapeo database" - but it was too fast to take note of and then it said sync complete, and seemed to be complete. I could not reproduce this. - not blocking
- After the sync there was an error message at bottom of screen, see photo below. I dismissed it. 
IMG_8897 - not blocking

**Comment by jencastrodoesstuff** [comment]
Gave RC830 a full test including install, load configurations, add, edit, share, delete, sync with Md &Mm.   Karissa did minor corrections to characters for share function that she sent me. 
There is that strange issue of data not appearing on the map for a while even though it is in the observation list. Tom identified it on his summary as well. This is a strange behavior I encountered with a guardaparque in the RCA but we couldn't figure out if it was time that made the points appear or connectivity to the internet.  Can figure out the sequence for testing this for the next release.

**Comment by jencastrodoesstuff** [comment]
What is the process for us to review apps that are ready to release?

**Review by jencastrodoesstuff — APPROVED** [review]
**Tested and passed***
Install
load config
online map loads
_*known issue: points don't appear in mapview when there is no online map and no offline map_
offline map loads
center to location
live track location on map
GPS, precision update and details screen
observation points appear on map with GPS OFF

new observation: select category, add description add photos, add details, save
edit observation: edit category, edit description, add photos, edit details, save
_*known issue: photos requires datestamp for verification purposes_
 manual entry GPS coordinates

Share with WhatsApp - custom formatting
Share with Email - custom subject line

Delete

Sync with Wifi
Syncing big databases from desktop* and mobile (900+ observations)
_*known issue: sync process termination and error front end issues_
Resyncing after errors (from previously interrupted syncs)

**PASS**
_Release the hounds!!_

**Review by okdistribute — APPROVED** [review]

**Pull request merged**
